nixos/modules/services/networking/uptermd.nix

```diff
@@ -0,0 +1,83 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.uptermd;
+in
+{
+  options = {
+    services.uptermd = {
+      enable = mkEnableOption "uptermd";
+
+      openFirewall = mkOption {
+        type = types.bool;
+        default = false;
+        description = "Whether to open the firewall for the port in <option>services.uptermd.port</option>.";
+      };
+
+      port = mkOption {
+        type = types.port;
+        default = 2222;
+        description = ''
+          Port the server will listen on.
+        '';
+      };
+
+      listenAddress = mkOption {
+        type = types.str;
+        default = "[::]";
+        example = "127.0.0.1";
+        description = ''
+          Address the server will listen on.
+        '';
+      };
+
+      extraFlags = mkOption {
+        type = types.listOf types.str;
+        default = [];
+        example = [ "--debug" ];
+        description = ''
+          Extra flags passed to the uptermd command.
+        '';
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    networking.firewall = mkIf cfg.openFirewall {
+      allowedTCPPorts = [ cfg.port ];
+    };
+
+    systemd.services.uptermd = {
+      description = "Upterm Daemon";
+      wantedBy = [ "multi-user.target" ];
+      after = [ "network.target" ];
+
+      serviceConfig = {
+        DynamicUser = true;
+        ExecStart = "${pkgs.upterm}/bin/uptermd --ssh-addr ${cfg.listenAddress}:${toString cfg.port} ${concatStringsSep " " cfg.extraFlags}";
```

Comment:
Add `upterm` to the `systemd.services.uptermd.path` variable in and just write `uptermd` here.

Reply:
@andir - I'm embarrassed to ask, but is including the package is `path` the preferred way? I see it done explicitly like here, or the way you suggest... Any benefits I'm not thinking of?

Reply:
I just think it is nicer as we do not have to do the string templating in ExecStart. It might be more precise but it is probably as precise as passing it explicitly.

The PATH for systemd units is already fairly restrictive and thus I see (almost) not downside in doing it this way.

String templating is a nice way to show that we can be explicit but lately I started seeing it as an anit-pattern.

Reply:
Thanks for your response. Maybe we start including `util-linux` in path for services which utilize `kill` in the `ExecReload` too... as a random example.

Reply:
> Thanks for your response. Maybe we start including `util-linux` in path for services which utilize `kill` in the `ExecReload` too... as a random example.

That might be good but we should be careful here not to have a list of 20 dependencies of which nobody knows why they are in there. Commenting on the attributes becomes more important this way.